### DISCUSSION TITLE
variable importance plot in ascending order

### DISCUSSION THREAD

**Discussion by suman762** [kogalur/randomForestSRC#338] [General]
Using the randomForestSRC  package, I can easily construct the variable importance plot, but can’t figure out how to report variables in ascending order (most important one first in the ordering).
For example, the following codes produce a plot with descending order of 10 variables (opposite of what I want)
oo <- subsample(vd.obj, B = 100)
plot.subsample(oo, pmax=10)
Suggestions please!

**Comment by ishwaran** [comment]
The importance vector is stored in the forest object.  Then you only need to sort the values in order.  

Here's an example with mtcars:

```
> o=rfsrc(mpg~.,mtcars,importance=TRUE)
> o$importance
        cyl        disp          hp        drat          wt        qsec          vs          am        gear        carb 
20.92994699 18.55855726 11.05339060  0.48036969 17.87492612  0.19481806  0.76215660  0.09743581  0.30144679  0.31365013 
> sort(o$importance)
         am        qsec        gear        carb        drat          vs          hp          wt        disp         cyl 
 0.09743581  0.19481806  0.30144679  0.31365013  0.48036969  0.76215660 11.05339060 17.87492612 18.55855726 20.92994699 
```

**Reply by suman762** [reply]
Thank s for your suggestions. I am still struggling how to present the sorted variables in the variable importance plot.  
The following codes can produce in descending order, but I need to reverse the order.
vd.obj <- rfsrc(Surv(time,status)~., vd,  block.size = 1,na.action = "na.impute")
oo <- subsample(vd.obj, B = 100)
plot.subsample(oo, pmax=10)

thanks for your help.

**Comment by ishwaran** [comment]
You will have to edit the `plot.subsample` function and modify it and create a new function in order to achieve that.  It should be relatively easy to do.

To see the` plot.subsample `function, simply type it at the terminal window:

`plot.subsample`

**Comment by suman762** [comment]
I updated the plot.subsample function with decreasing = FALSE, but see the following error when running the line plot.subsample(oo, pmax=10).
Error in get.univariate.target(x$rf, m.target) : 
  could not find function "get.univariate.target"  

I do not see any function "get.univariate.target" in the package. Please help if I miss something. Thanks.

**Reply by kogalur** [reply]
That's not a publically exported function. You will need to look at the source code in the package and duplicate it in your script.

**Comment by suman762** [comment]
I imported the .tar.gz file from the CRAN and then updated the function, but then observed the error I mentioned above that I don't know how to resolve. Suggestions please.

**Reply by ishwaran** [reply]
No this is not the right way to solve this issue.  There is a simple workaround that is used in R using the double colon.  See my next post.

**Comment by ishwaran** [comment]
Use `randomForestSRC::get.univariate.target`

See the following stackoverflow discussion for explanation of what double colon :: in R does

https://stackoverflow.com/questions/35240971/what-are-the-double-colons-in-r

**Reply by suman762** [reply]
I run all the functions from https://rdrr.io/cran/randomForestSRC/src/R/utilities.multivariate.R 
and then updated the function plot.subsample to solve the problem. 
I have a suggestion: I think it would be handy to update the function so that user can order the plot either in ascending or descending order.

**Comment by suman762** [comment]
I'm trying to understand your suggestions. Get an idea of purpose of using double colons, but the error suggests the "get.univariate.target" does not exists.  How the function "plot.subsample" should be updated to get the desired plot? When I used your suggestions of using randomForestSRC::get.univariate.target, I see the following error.
Error: 'get.univariate.target' is not an exported object from 'namespace:randomForestSRC'

Did I miss something basic or it is a real bug which can be solved with a toy example. Thanks.

**Comment by ishwaran** [comment]
In the next version of the package, we will make it possible to sort the order of the plot.
